### ISSUE TITLE
Quick Question Negative/Inverted Images

### ISSUE THREAD

**Issue by dparres** [FactoDeepLearning/DAN#10]
Hello!

I have a question about the treatment of images in training! I noticed that the "_getitem_()" images are "normal". But when they arrive to the OCRCollateFunction "__call__()", these images have been negativized or inverted. How can I deactivate the negative/inverted filter to train with the "normal" image? I can't find in which line of code it is applied, as I thought the code flow of the training loop is __getitem__() -> collate __call__() -> train_batch. But I seem to be missing something...

Best regards and thank you very much,
Daniel

**Comment by FactoDeepLearning** [comment]
Hello,

This may be due to data augmentation, you can disable it by setting 
"augmentation": None 
in the main file.

Or it is due to the normalization, you need to denormalize the images to display them correctly.

**Comment by dparres** [comment]
Despite disabling normalization and augmentation in the main file the negative/inverted filter is applied. And I think it's because I don't understand the code flow. Because in __getitem__() the image is correct, but when I visualize the image in the collate function __call__() it already has the negative filter applied...

__getitem__() -> ? -> collate __call__() -> train_batch

I don't know which function or code would be "?" where this filter is applied, because I can't find it...

**Comment by FactoDeepLearning** [comment]
Can you provide a visualization of the problem, and the exact line of the code from which you display the image, please

**Comment by dparres** [comment]
In trainer_dan.py I print the first image of the batch and I obtain the following inverted image. 
![image](https://user-images.githubusercontent.com/114649578/217214075-d208866c-a89b-4b7d-8ea6-0d68705d8fca.png)
![debug_train_batch_normal](https://user-images.githubusercontent.com/114649578/217213906-bd913bfb-52dc-4ff7-8ceb-39be28a13377.jpg)

**Comment by FactoDeepLearning** [comment]
and what are the min and max you computed ?
This may be an encoding issue between float tensor and PIL, have you tried using matplotlib for display ?

**Comment by dparres** [comment]
Yes! Is a issue between Torch and PIL... With matplotlib everything works perfectly! Thank you very much!
![image](https://user-images.githubusercontent.com/114649578/217224363-38fc0611-bc03-471a-8478-6fc37ca4549b.png)
![image](https://user-images.githubusercontent.com/114649578/217227703-5b48b848-8a7b-452f-9ce5-6659bd9d6b4f.png)
